src/odemis/gui/comp/overlay/cryo_feature.py

```diff
@@ -380,3 +387,420 @@ def _get_feature_position_at_view_posture(self, feature: CryoFeature) -> Dict[st
+                self.cnvs.set_dynamic_cursor(gui.DRAG_CURSOR)
+                DragMixin._on_motion(self, evt)
+                p_pos = self.cnvs.view_to_phys(v_pos, self.cnvs.get_half_buffer_size())
+                self._selected_target.coordinates.value = [p_pos[0], p_pos[1],
+                                                           self._selected_target.coordinates.value[2]]
+                self.cnvs.update_drawing()
+                return
+            target = self._detect_point_inside_target(v_pos)
+            if target:
+                return
+            else:
+                if self._mode == MODE_EDIT_FIDUCIALS or self._mode == MODE_EDIT_POI:
+                    self.cnvs.set_default_cursor(wx.CURSOR_PENCIL)
+                else:
+                    return
+        WorldOverlay.on_motion(self, evt)
+
+    def draw(self, ctx, shift=(0, 0), scale=1.0):
+        """
+        Draw all the targets, on their location, indicating their status.
+        """
+        if not self.show:
+            return
+
+        # Show each target icon and label if applicable
+        for target in self.tab_data.main.targets.value:
+            if "FM" in target.name.value or "POI" in target.name.value:
+                coordinates = target.coordinates.value
+                half_size_offset = self.cnvs.get_half_buffer_size()
+
+                # convert physical position to buffer 'world' coordinates
+                bpos = self.cnvs.phys_to_buffer_pos((coordinates[0], coordinates[1]), self.cnvs.p_buffer_center, self.cnvs.scale,
+                                                    offset=half_size_offset)
+                def set_icon(feature_icon):
+                    ctx.set_source_surface(feature_icon, bpos[0] - FEATURE_ICON_CENTER, bpos[1] - FEATURE_ICON_CENTER)
+
+                # Show proper feature icon based on selected target + status
+                try:
+                    if target is self.tab_data.main.currentTarget.value:
+                        # Correct label positions such that label is outside the icon display
+                        set_icon(self._feature_icons_selected[target.type.value])
+                        self._label.text = target.name.value
+                        self._label.pos = (bpos[0]+10, bpos[1]+10)
+                        self._label.draw(ctx)
+                    elif self.tab_data.main.currentTarget.value and (target.index.value == self.tab_data.main.currentTarget.value.index.value) and ("FIB" in self.tab_data.main.currentTarget.value.name.value) and ("POI" not in target.name.value):
+                        set_icon(self._feature_icons_selected["FiducialPair"])
+                    else:
+                        set_icon(self._feature_icons[target.type.value])
+                except KeyError:
+                    raise
+
+                ctx.paint()
+
+
+class CryoCorrelationFibPointsOverlay(CryoCorrelationPointsOverlay):
+
+    def _detect_point_inside_target(self, v_pos):
```

Comment:
This can become generic in the base class
```suggestion
    def _detect_point_inside_target(self, v_pos, allowed_target_type: set):
```

then in line 641
```
if target.type.value in allowed_target_type:
```